src/status_im/utils/ethereum/eip681.cljs

```diff
@@ -94,5 +95,11 @@
              (str chain-id-separator chain-id))
            (when-not (empty? parameters)
              (if function-name
-               (str function-name-separator function-name query-separator (generate-query-string function-arguments))
-               (str query-separator (generate-query-string parameters))))))))
\ No newline at end of file
+               (str function-name-separator function-name query-separator (generate-query-string (merge function-arguments (dissoc parameters :function-name :function-arguments))))
```

Comment:
hard to read, break up with let/threading

Reply:
:+1: